src_c/_sdl2/video.pyx

```diff
@@ -940,3 +977,110 @@ cdef class Renderer:
         if not dest:
             return self.get_viewport()
         return dest
+
+    def draw_line(self, p1, p2):
+        # https://wiki.libsdl.org/SDL_RenderDrawLine
+        res = SDL_RenderDrawLine(self._renderer,
+                                 p1[0], p1[1],
+                                 p2[0], p2[1])
+        if res < 0:
+            raise error()
+
+    def draw_point(self, point):
+        # https://wiki.libsdl.org/SDL_RenderDrawPoint
+        res = SDL_RenderDrawPoint(self._renderer,
+                                  point[0], point[1])
+        if res < 0:
+            raise error()
+
+    def draw_rect(self, rect):
+        # https://wiki.libsdl.org/SDL_RenderDrawRect
+        cdef SDL_Rect _rect
+        cdef SDL_Rect *rectptr = pgRect_FromObject(rect, &_rect)
+        if rectptr == NULL:
+            raise TypeError('expected a rectangle')
+        res = SDL_RenderDrawRect(self._renderer, rectptr)
+        if res < 0:
+            raise error()
+
+    def fill_rect(self, rect):
+        # https://wiki.libsdl.org/SDL_RenderFillRect
+        cdef SDL_Rect _rect
+        cdef SDL_Rect *rectptr = pgRect_FromObject(rect, &_rect)
+        if rectptr == NULL:
+            raise TypeError('expected a rectangle')
+        res = SDL_RenderFillRect(self._renderer, rectptr)
+
+        if res < 0:
+            raise error()
+
+    def to_surface(self, surface=None, area=None):
+        # https://wiki.libsdl.org/SDL_RenderReadPixels
+        """
+            Read pixels from the current rendering target and create a pygame.Surface.
+            WARNING: This is a very slow operation, and should not be used frequently.
+
+        :param surface: A surface to read the pixel data into.
+                        It must be large enough to fit the area, or ``ValueError`` is
+                        raised.
+                        If ``None``, a new surface is returned.
+        :param area: The area of the screen to read pixels from. The area is
+                     clipped to fit inside the viewport.
+                     If ``None``, the entire viewport is used.
+        """
+        cdef Uint32 format
+        cdef SDL_Rect rarea
+        cdef SDL_Rect tempviewport
+        cdef SDL_Rect *areaparam
+        cdef SDL_Surface *surf
+        cdef SDL_Texture *targettex
+
+        # obtain area to use
+        if area is not None:
+            if pgRect_FromObject(area, &rarea) == NULL:
+                raise TypeError('area must be None or a rect')
+
+            # clip area
+            SDL_RenderGetViewport(self._renderer, &tempviewport)
+            SDL_IntersectRect(&rarea, &tempviewport, &rarea)
```

Comment:
What should happen if there is no intersection?

Reply:
The current behavior is probably right: Return a new 0x0 surface if no surface is passed, or else just return the passed-in surface.